Jenkinsfile

```diff
@@ -141,69 +178,76 @@ def uploadCli(architectures) {
         def currentImage = images[i]
         def primaryName = currentImage.names[0]
 
-        buildDockerImage(primaryName, version, currentImage.dockerFile, jfrogCliRepoDir)
-        pushDockerImageVersionAndRelease(primaryName, version)
+        def imageRepo21Name = "$repo21Prefix/$primaryName"
+        buildDockerImage(imageRepo21Name, version, currentImage.dockerFile, jfrogCliRepoDir)
+        pushDockerImageVersion(imageRepo21Name, version)
 
         // Push alternative tags if needed.
         for (int n = 1; n < currentImage.names.size(); n++) {
             def newName = currentImage.names[n]
-            // Create new tag.
-            sh """#!/bin/bash
-                docker tag $primaryName:$version $newName:$version
-            """
-            pushDockerImageVersionAndRelease(newName, version)
+            def currentRepo21Name = "$repo21Prefix/$newName"
+            buildDockerImage(currentRepo21Name, version, currentImage.dockerFile, jfrogCliRepoDir)
+            pushDockerImageVersion(currentRepo21Name, version)
         }
     }
-}
 
-def buildDockerImage(name, version, dockerFile, jfrogCliRepoDir) {
-    dir("$jfrogCliRepoDir") {
-        sh """#!/bin/bash
-            docker build --tag=$name:$version -f $dockerFile .
-        """
+    stage("Distribute cli-docker-images to releases") {
+        distributeToReleases("cli-docker-images2", version, "docker-images-rbc-spec.json")
+    }
+
+    stage("Promote docker images") {
+        for (int i = 0; i < images.size(); i++) {
+            def currentImage = images[i]
+            promoteDockerImage(currentImage.name, version, jfrogCliRepoDir)
+        }
     }
 }
 
-def pushDockerImageVersionAndRelease(name, version) {
+def promoteDockerImage(name, version, jfrogCliRepoDir) {
+    print "Promoting docker image: $name"
     withCredentials([string(credentialsId: 'jfrog-cli-automation', variable: 'JFROG_CLI_AUTOMATION_ACCESS_TOKEN')]) {
         options = "--url https://releases.jfrog.io/artifactory --access-token=$JFROG_CLI_AUTOMATION_ACCESS_TOKEN"
         sh """#!/bin/bash
-            builder/jfrog rt docker-push $name:$version reg2 $options
-            docker tag $name:$version $name:latest
-            builder/jfrog rt docker-push $name:latest reg2 $options
-        """
+            $builderPath rt docker-promote $name reg2 reg2 --copy --source-tag=$version --target-tag=latest $options
+            """
+        }
     }
 }
 
-def uploadGetCliToJfrogReleases() {
-    withCredentials([string(credentialsId: 'jfrog-cli-automation', variable: 'JFROG_CLI_AUTOMATION_ACCESS_TOKEN')]) {
+def buildDockerImage(name, version, dockerFile, jfrogCliRepoDir) {
+    dir("$jfrogCliRepoDir") {
         sh """#!/bin/bash
-                builder/jfrog rt u $jfrogCliRepoDir/build/getCli.sh jfrog-cli/v1/scripts/ --url https://releases.jfrog.io/artifactory/ --access-token=$JFROG_CLI_AUTOMATION_ACCESS_TOKEN --flat
+            docker build --tag=$name:$version -f $dockerFile .
```

Comment:
The Dockerfiles in v2 were modified so that images will only be pulled from repo21. The change is missing here, and once done, the docker build command should be modified accordingly as well.